### PR TITLE
[20 minute tasks] Add test for when a batch is full

### PR DESCRIPTION
**Pull request by tboam**
**Goals (and why)**: Add tests to codify behaviour.  In particular make clear the fact that we keep building a batch until we've finished a row which some people have questioned before.

**Implementation Description (bullets)**: Wrote the test

**Concerns (what feedback would you like?)**: My verbose BDD test writing style is sure to come under attack

**Where should we start reviewing?**: In the test

**Priority (whenever / two weeks / yesterday)**: 20 minutes!

---
This change is [Reviewable](https://reviewable.io/reviews/palantir/atlasdb/2655)

### CODE REVIEW

File: atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/ExaminedCellLimitTest.java [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+        currentRow = curRow;
+    }
+
+    //Then
+
+    private void examinedEnoughCellsIs(boolean expected) {
+        CellsToSweepPartitioningIterator.ExaminedCellLimit limit
+                = new CellsToSweepPartitioningIterator.ExaminedCellLimit(startingRow.getBytes(), maxCellsToExamine);
+
+        assertThat(limit.examinedEnoughCells(cellsExamined, cell()), Matchers.is(expected));
```
Comment (jeremyk-91) [inline atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/ExaminedCellLimitTest.java]: nit: standard practice is static import

File: atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/ExaminedCellLimitTest.java [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+        //When
+        examined(1);
+        on("row2");
+
+        //Then
+        examinedEnoughCellsIs(false);
+    }
+
+    @Test
+    public void whenNextRowAndAtLimit_thenExaminedEnoughIsFalse() {
```
Comment (jeremyk-91) [inline atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/ExaminedCellLimitTest.java]: This should read true 😄

Comment (tboam) [reply atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/ExaminedCellLimitTest.java]: Good point, thanks

File: atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/ExaminedCellLimitTest.java (hunk 1 above)
Comment (tboam) [reply atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/ExaminedCellLimitTest.java]: Done

**Comment by hsaraogi** [comment]
I dont get how code coverage says this will decrease coverage :(

**Comment by tboam** [comment]
@hsaraogi that's a very good question!  My code change is nowhere near those classes.  I suspect that sometimes our tests throw an exception and retry (e.g. waiting for a leader).  So it is non-deterministic whether some code paths get executed on a certain run.

This raises the question of whether the code coverage metric is useful

**Pull request merged**